fuel-core/src/database.rs

```diff
@@ -377,125 +377,17 @@ mod relayer {
-        ops::DerefMut,
-        sync::Arc,
+        relayer::RelayerDb,
     };
+    use std::sync::Arc;
 
     #[async_trait::async_trait]
     impl RelayerDb for Database {
-        async fn get_validators(
-            &self,
-        ) -> HashMap<ValidatorId, (ValidatorStake, Option<ConsensusId>)> {
-            struct WrapAddress(pub ValidatorId);
-            impl From<Vec<u8>> for WrapAddress {
-                fn from(i: Vec<u8>) -> Self {
-                    Self(ValidatorId::try_from(i.as_ref()).unwrap())
-                }
-            }
-            let mut out = HashMap::new();
-            for diff in self
-                .iter_all::<WrapAddress, (ValidatorStake, Option<ConsensusId>)>(
-                    Column::ValidatorsSet,
-                    None,
-                    None,
-                    None,
-                )
-            {
-                match diff {
-                    Ok((address, stake)) => {
-                        out.insert(address.0, stake);
-                    }
-                    Err(err) => panic!("Database internal error:{:?}", err),
-                }
-            }
-            out
-        }
-
-        async fn get_staking_diffs(
-            &self,
-            from_da_height: DaBlockHeight,
-            to_da_height: Option<DaBlockHeight>,
-        ) -> Vec<(DaBlockHeight, StakingDiff)> {
-            let to_da_height = if let Some(to_da_height) = to_da_height {
-                if from_da_height > to_da_height {
-                    return Vec::new()
-                }
-                to_da_height
-            } else {
-                DaBlockHeight::from(u64::MAX)
-            };
-            struct WrapU64Be(pub DaBlockHeight);
-            impl From<Vec<u8>> for WrapU64Be {
-                fn from(i: Vec<u8>) -> Self {
-                    use byteorder::{
-                        BigEndian,
-                        ReadBytesExt,
-                    };
-                    use std::io::Cursor;
-                    let mut i = Cursor::new(i);
-                    Self(DaBlockHeight::from(
-                        i.read_u64::<BigEndian>().unwrap_or_default(),
-                    ))
-                }
-            }
-            let mut out = Vec::new();
-            for diff in self.iter_all::<WrapU64Be, StakingDiff>(
-                Column::StakingDiffs,
-                None,
-                Some(from_da_height.to_be_bytes().to_vec()),
-                None,
-            ) {
-                match diff {
-                    Ok((key, diff)) => {
-                        let block = key.0;
-                        if block > to_da_height {
-                            return out
-                        }
-                        out.push((block, diff))
-                    }
-                    Err(err) => panic!("get_validator_diffs unexpected error:{:?}", err),
-                }
-            }
-            out
-        }
-
-        async fn apply_validator_diffs(
```

Comment:
Replying to @xgreenx about old data cleanup:

Yes, we should hunt down the remaining places this unused data is being managed and clean it up.

Reply:
TODO: Make issue for this